src/basic/process-util.c

```diff
@@ -312,19 +312,18 @@ int rename_process(const char name[]) {
         /* Third step, completely replace the argv[] array the kernel maintains for us. This requires privileges, but
          * has the advantage that the argv[] array is exactly what we want it to be, and not filled up with zeros at
          * the end. This is the best option for changing /proc/self/cmdline. */
-        if (mm_size < l+1) {
+
+        /* Let's not bother with this if we don't have euid == 0. Strictly speaking if people do weird stuff with
+         * capabilities this could work even for euid != 0, but our own code generally doesn't do that, hence let's
+         * use this as quick bypass check, to avoid calling mmap() if PR_SET_MM_ARG_{START,END} fails with EPERM later
+         * on anyway. After all geteuid() is dead cheap to call, but mmap() is not. */
+        if (geteuid() != 0) {
```

Comment:
Anyway, the PR fixes the real life issue, so I think it can be merged as is after removing the braces and the empty line.